tests/components/feedreader/test_config_flow.py

```diff
@@ -0,0 +1,231 @@
+)
+from homeassistant.config_entries import SOURCE_RECONFIGURE, SOURCE_USER
+from homeassistant.const import CONF_URL
+from homeassistant.core import DOMAIN as HA_DOMAIN, HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+from homeassistant.helpers import issue_registry as ir
+from homeassistant.setup import async_setup_component
+
+from .const import FEED_TITLE, URL, VALID_CONFIG_DEFAULT
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture(name="feedparser")
+def feedparser_fixture(feed_one_event: bytes) -> Mock:
+    """Patch libraries."""
+    with (
+        patch(
+            "homeassistant.components.feedreader.config_flow.feedparser.http.get",
+            return_value=feed_one_event,
+        ) as feedparser,
+    ):
+        yield feedparser
+
+
+@pytest.fixture(name="setup_entry")
+def fsetup_entry_fixture(feed_one_event: bytes) -> Mock:
+    """Patch libraries."""
+    with (
+        patch("homeassistant.components.feedreader.async_setup_entry") as setup_entry,
+    ):
+        yield setup_entry
+
+
+async def test_user(hass: HomeAssistant, feedparser, setup_entry) -> None:
+    """Test starting a flow by user."""
+    # init user flow
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    # success
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_URL: URL, CONF_MAX_ENTRIES: 5}
+    )
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    assert result["title"] == FEED_TITLE
+    assert result["data"][CONF_URL] == URL
+    assert result["data"][CONF_MAX_ENTRIES] == 5
+
+
+async def test_import(hass: HomeAssistant, feedparser, setup_entry) -> None:
+    """Test starting an import flow."""
+    config_entries = hass.config_entries.async_entries(DOMAIN)
+    assert not config_entries
+
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_URLS: [URL]}})
+
+    config_entries = hass.config_entries.async_entries(DOMAIN)
+    assert config_entries
+    assert len(config_entries) == 1
+    assert config_entries[0].title == FEED_TITLE
+    assert config_entries[0].data[CONF_URL] == URL
+    assert config_entries[0].data[CONF_MAX_ENTRIES] == DEFAULT_MAX_ENTRIES
+
+    assert ir.async_get(hass).async_get_issue(HA_DOMAIN, "deprecated_yaml_feedreader")
+
+
+async def test_user_errors(
+    hass: HomeAssistant, feedparser, setup_entry, feed_one_event
+) -> None:
+    """Test starting a flow by user which results in an URL error."""
+    # init user flow
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    # raise URLError
+    feedparser.side_effect = urllib.error.URLError("Test")
+    feedparser.return_value = None
```

Comment:
I don't think we have to set `feedparser.return_value` to `None` twice

Reply:
this is more about to be explicit and having a common design of the tests (_first set both mock results, than test these results_)